File: api/Services/LogsService.cs
using EmpresaApi.Data;
using EmpresaApi.Interfaces;
using EmpresaApi.DTOs;
using EmpresaApi.Responses;
using Microsoft.EntityFrameworkCore;

namespace EmpresaApi.Services
{
    public class LogsService : ILogsService
    {
        private readonly EmpresaContext _context;

        public LogsService(EmpresaContext context)
        {
            _context = context;
        }

        public async Task<ApiResponse<List<LogTransacionDTO>>> ObtenerTodosAsync(int page = 1, int pageSize = 50)
        {
            var logs = await _context.Logs
                .OrderByDescending(l => l.Fecha)
                .Skip((page - 1) * pageSize)
                .Take(pageSize)
                .Select(l => new LogTransacionDTO
                {
                    Id = l.Id,
                    VerboHttp = l.VerboHttp,
                    Endpoint = l.Endpoint,
                    StatusCode = l.StatusCode,
                    Payload = l.Payload,
                    Fecha = new DateTimeOffset(DateTime.SpecifyKind(l.Fecha, DateTimeKind.Utc))
                })
                .ToListAsync();

            return ApiResponse<List<LogTransacionDTO>>.SuccessResponse(logs);
        }

        public async Task<ApiResponse<LogTransacionDTO>> ObtenerPorIdAsync(int id)
        {
            var log = await _context.Logs.FindAsync(id);
            if (log == null)
                return ApiResponse<LogTransacionDTO>.ErrorResponse("Log no encontrado", 404);

            return ApiResponse<LogTransacionDTO>.SuccessResponse(new LogTransacionDTO
            {
                Id = log.Id,
                VerboHttp = log.VerboHttp,
                Endpoint = log.Endpoint,
                StatusCode = log.StatusCode,
                Payload = log.Payload,
                Fecha = new DateTimeOffset(DateTime.SpecifyKind(log.Fecha, DateTimeKind.Utc))
            });
        }
    }
}

File: frontend/src/services/logs.service.ts
import api, { ApiResponse } from '@/lib/api'

export interface LogTransacionDTO {
    id: number
    verboHttp: string
    endpoint: string
    statusCode: number
    payload: string
    fecha: string
}

export const logsService = {
    obtenerTodos: async (page: number = 1, pageSize: number = 50): Promise<LogTransacionDTO[]> => {
        const response = await api.get(`/logs?page=${page}&pageSize=${pageSize}`) as ApiResponse<LogTransacionDTO[]>
        if (!response.success) throw new Error(response.message)
        return response.data
    },

    obtenerPorId: async (id: number): Promise<LogTransacionDTO> => {
        const response = await api.get(`/logs/${id}`) as ApiResponse<LogTransacionDTO>
        if (!response.success) throw new Error(response.message)
        return response.data
    }
}
